### ISSUE TITLE
Where does the trajectory vocabulary come from?

### ISSUE THREAD

**Issue by yihongXU** [woxihuanjiangguo/ZTRS#4]
Hi, 

Very nice work, could you have more details about the trajectory vocabulary? How is it generated and collected?

Thanks!

**Comment by woxihuanjiangguo** [comment]
Hi @yihongXU ,

We first sampled 700K trajectories (x, y, heading, 4 seconds, 10Hz) from the nuPlan dataset. Then we run K-Means clustering on these trajectories with:
```
from sklearn.cluster import KMeans

vocab_size = 16384
L, HORIZON, DIM = ori_traj.shape
all_traj = ori_traj.reshape(L, -1)
clustering = KMeans(
        vocab_size,
        verbose=True,
        tol=0.0,
        init="k-means++",
        n_init="auto",
).fit(all_traj)
anchors = clustering.cluster_centers_.reshape(vocab_size, HORIZON, DIM)

**Comment by yihongXU** [comment]
Thank you for the detailed answer, thanks for sharing this insightful work!!
